### PR TITLE
ARROW-7054: [Docs] Enable overriding project version with environment variable when building Sphinx docs

### PR DESCRIPTION
**Pull request by wesm**
This is a workaround for ARROW-7053, but it can also help with adding new sections to the docs in between releases when needed.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -110,9 +110,11 @@
 # built documents.
 #
 # The short X.Y version.
-version = pyarrow.__version__
+version = os.environ.get('ARROW_DOCS_SPOOFED_VERSION',
```
Comment (kou) [inline docs/source/conf.py]: How about just using `ARROW_DOCS_VERSION`?

**Review by kou — APPROVED** [review]
+1

File: docs/source/conf.py (hunk 1 above)
Comment (pitrou) [reply docs/source/conf.py]: +1

Comment (wesm) [reply docs/source/conf.py]: OK

**Pull request closed without merging**